pkg/systemd/overview-cards/cryptoPolicies.jsx

```diff
@@ -89,18 +101,34 @@ const CryptoPolicyDialog = ({
     setCurrentCryptoPolicy,
 }) => {
     const [error, setError] = useState();
+    const [inProgress, setInProgress] = useState(undefined);
     const [selected, setSelected] = useState(currentCryptoPolicy);
+    const [rebootRequired, setRebootRequired] = useState(currentCryptoPolicy);
 
     const setPolicy = (reboot) => {
+        const applyPolicy = reboot => {
+            if (reboot) {
+                cockpit.spawn(["shutdown", "--reboot", "now"], { superuser: "require", err: "message" })
+                        .catch(error => setError(error));
+            } else {
+                cockpit.file(applyNeedRebootStamp, { superuser: "require" }).replace("\n")
+                        .then(() => close());
+            }
+        };
         cockpit.spawn(["update-crypto-policies", "--set", selected], { superuser: "require", err: "message" })
                 .then(() => {
                     setCurrentCryptoPolicy(selected);
-                    if (reboot) {
-                        cockpit.spawn(["shutdown", "--reboot", "now"], { superuser: "require", err: "message" })
-                                .catch(error => setError(error));
+                    if (selected === "FIPS") {
+                        setRebootRequired(reboot);
+                        setInProgress(true);
+                        cockpit.spawn(["fips-mode-setup", "--enable"], { superuser: "require", err: "message" })
```

Comment:
To make double sure: just calling `fips-mode-setup --enable` will *not* automatically switch the crypto policy to "FIPS"? This still has to happen explicitly? The [documentation](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/security_hardening/using-the-system-wide-cryptographic-policies_security-hardening#switching-the-system-to-fips-mode_using-the-system-wide-cryptographic-policies) specifically documents "FIPS" as

> A policy level that conforms with the FIPS 140-2 requirements. This is used internally by the fips-mode-setup tool, which switches the RHEL system into FIPS mode. 

So ISTM we should call one or the other, but not both.